### ISSUE TITLE
SCSS '@import' discouraged in favour of '@use'

### ISSUE THREAD

**Issue by arrontp** [csalmeida/hozokit#17]
SASS/SCSS are discouraging and phasing out support for `@import` rule in favour of the `@use` rule.

This makes styling more readable, safer and reduces compile time.

For more detail see '⚠**Heads Up!**' notice in the [documentation here](https://sass-lang.com/documentation/at-rules/import).

Read [`@use` rule docs here](https://sass-lang.com/documentation/at-rules/use).

**Comment by csalmeida** [comment]
Awesome, thanks for pointing it out. 🙏 Hozokit does have a few `@import` rules here and there (around 10).
I can't investigate right now but I'll have a look as soon as I have a spare moment.

If you would like to have a look in the meantime:

- Check if the [node-sass package](https://www.npmjs.com/package/node-sass) we use relies on Dart Sass. If it does it means that we can replace all `@import` statements with `@use`.
- Replace all `@import` with `@use` and run a test version to check if styles still work as expected. 

PR is welcome of course!

**Comment by arrontp** [comment]
I'm already on it! 😉

I'll aim to have a PR done by Sunday.

**Comment by arrontp** [comment]
⚠ **'node-sass' is depreciated.**
> "While they will continue to receive maintenance releases indefinitely, there are no plans to add additional features or compatibility with any new CSS or Sass features. Projects that still use it should move onto Dart Sass."

Our current compiler doesn't support the latest features of Sass and therefore doesn't recognise the new `@use` rule.
We should move to introduce **Dart Sass** as recommended by [node-sass](https://github.com/sass/node-sass#options), [gulp-sass](https://www.npmjs.com/package/gulp-sass) and [Sass in general](https://sass-lang.com/dart-sass).

I'll update the `gulpfile.js` to be explicit that we want Dart Sass.
```
sass.compiler = require('sass');

**Comment by arrontp** [comment]
In the Hozokit Gulp script, why do we concatenate the styles into a `temp.scss` file before compile?
```css
/* Compiles a style.css for the front facing side of the site. */
gulp.task('styles', () => {
  return gulp.src(stylesheetCompilePaths)  
    .pipe(concat({path: './styles/temp.scss'})) /* <<< HERE */
    .pipe(sass().on('error', sass.logError))
    .pipe(gulpif( envIsNotDevelopment || minifyEnabled, cleanCSS() ))
    .pipe(rename('style.css'))
    .pipe(gulp.dest('./'))
})
```
----
Because the issue I'm running into is that compiling will now throw an error as the `@use` rule must be declared before anything else at the top of a scss file.
But maybe I'm missing something 🤔

**Comment by csalmeida** [comment]
That's a good solution but perhaps in the long term we should switch package since I noticed that `gulp-sass` hasn't switched to Dart. One that could be used is https://github.com/Mister-Hope/gulp-sass.

However, if you think `require('sass')` is doing the job correctly then we can keep it!

As for the temp file, if I recall correctly I had to create a file temporarily to make sure variables were available to component style files at compile time, it was a long time ago so maybe we can find a better approach for it. 

If you remove `.pipe(concat({path: './styles/temp.scss'}))` you'll notice you will get errors related to undefined variables:

```bash
Message:
    templates/components/example/style.scss
Error: Undefined variable: "$primary-font".
        on line 3 of templates/components/example/style.scss
>>     font-family: $primary-font;
   -----------------^
```

Let's chat about it when we catch up. 🙏
